core/src/com/fallenflame/game/LevelController.java

```diff
@@ -1,5 +1,498 @@
+     * wireframes onscreen
+     *
+     * @return whether this level is currently in debug node
+     */
+    public boolean getDebug() { return debug; }
+
+    /**
+     * Sets whether this level is currently in debug node
+     *
+     * If the level is in debug mode, then the physics bodies will all be drawn as
+     * wireframes onscreen
+     *
+     * @param value	whether this level is currently in debug node
+     */
+    public void setDebug(boolean value) { debug = value; }
+
+    /**
+     * Returns the maximum FPS supported by this level
+     *
+     * This value is used by the rayhandler to fix the physics timestep.
+     *
+     * @return the maximum FPS supported by this level
+     */
+    public int getMaxFPS() { return maxFPS; }
+
+    /**
+     * Sets the maximum FPS supported by this level
+     *
+     * This value is used by the rayhandler to fix the physics timestep.
+     *
+     * @param value the maximum FPS supported by this level
+     */
+    public void setMaxFPS(int value) { maxFPS = value; }
+
+    /**
+     * Returns the minimum FPS supported by this level
+     *
+     * This value is used by the rayhandler to fix the physics timestep.
+     *
+     * @return the minimum FPS supported by this level
+     */
+    public int getMinFPS() { return minFPS; }
+
+    /**
+     * Sets the minimum FPS supported by this level
+     *
+     * This value is used by the rayhandler to fix the physics timestep.
+     *
+     * @param value the minimum FPS supported by this level
+     */
+    public void setMinFPS(int value) { minFPS = value; }
+
+    /**
+     * Gets the current state of the level.
+     *
+     * This value is used by GameEngine to know when player has lost or won.
+     */
+    public LevelState getLevelState() { return levelState; }
+
+    /**
+     * Sets the current state of the level.
+     *
+     * This value is used by GameEngine to know when player has lost or won.
+     *
+     * @param state Desired levelState
+     */
+    public void setLevelState(LevelState state) { levelState = state; }
+
+    /**
+     * Creates a new LevelModel
+     *
+     * The level is empty and there is no active physics world.  You must read
+     * the JSON file to initialize the level
+     */
+    public LevelController() {
+        // World
+        world  = null;
+        bounds = new Rectangle(0,0,1,1);
+        scale = new Vector2(1,1);
+        debug  = false;
+        levelState = LevelState.IN_PROGRESS;
+        // Controllers
+        lightController = new LightController();
+        AIControllers = new LinkedList<>();
+        // Models
+        walls = new LinkedList<>();
+        enemies = new LinkedList<>();
+        flares = new LinkedList<>();
+        levelModel = new LevelModel();
+    }
+
+    /**
+     * Lays out the game geography and enemies from the given JSON file
+     *
+     * @param levelFormat	the JSON tree defining the level
+     */
+    public void populate(JsonValue levelFormat) {
```

Comment:
For consistency's sake (also super picky), do we want to call our level JSON levelJson or levelFormat? I made GameEngine call it levelJson cause the other JSON files had that type of naming